src/model.py

```diff
@@ -73,17 +91,105 @@ def __init__(self):
                 Bucket=s3_bucket, Key=key, Filename=str(file_downloaded_path)
             )
 
-        loaded_model = tf.keras.models.load_model(
-            f"{trained_model_path}/{model_version}", compile=False
-        )
+        if USE_NEURAL_MAGIC:
+            onnx_filepath = f"{trained_model_path}/{model_version}.onnx"
+            batch_size = 1
+
+            # Compile
+            loaded_model = compile_model(onnx_filepath, batch_size)
+
+        elif USE_PYTORCH:
+            # The NN needs to be redefined
+            # MNIST dataset parameters.
+            num_classes = 10  # total classes (0-9 digits).
+
+            class CNN(nn.Module):
```

Comment:
Do you want to make this class independent of the class `Model` so that its not defined conditionally?

Reply:
Sure it can improved definetly :)